packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -244,7 +244,8 @@ class ScaffoldMessengerState extends State<ScaffoldMessenger> with TickerProvide
 
   // SNACKBAR API
 
-  /// Shows a [SnackBar] across all registered [Scaffold]s.
+  /// Shows a [SnackBar] across all registered [Scaffold]s. Each registered
```

Comment:
I wonder if we need to add this to the ScaffoldMessenger, Scaffold and SnackBar classes, in addition to `showSnackBar`.

Reading this on its own, I wonder, what are registered Scaffolds? Do we document that anywhere?

Reply:
> I wonder if we need to add this to the ScaffoldMessenger, Scaffold and SnackBar classes, in addition to showSnackBar

I first wondered that too but reached the conclusion that `showSnackBar` is a central place for this comment:
- `Scaffold` documentation has a lot of content and `SnackBar` is not mentioned there.
- `ScaffoldMessenger` quickly emphasizes the `showSnackBar` method so it made sense 
to focus on `showSnackBar` documentation.

But revisiting this, I see several possible improvements:
- As you mentioned, "registered Scaffolds" could be more explicit (this is explained in `ScaffoldMessengerState` documentation).
- `showMaterialBanner` could be improved in a similar way.
- `SnackBar` (and `MaterialBanner`) are not mentioned at all in `Scaffold` documentation. Adding a reference to them in the ‘See also’ section makes sense.
- Stating that each scaffold has its own instance of `SnackBar` is not fully correct. There is only one instance of `SnackBar` (the one passed to `showSnackBar`). As it is a stateful widget and it is inserted into the tree in multiple locations, the framework will create a separate `_SnackBarState` instance for each scaffold. I’m not sure about how to explain this in a comprehensive way. 

I updated the PR with those changes but the wording could probably be improved.